### PR TITLE
refactor(tests): consolidate TestReadFile with fixtures

### PR DESCRIPTION
**Pull request by major**
- Add CONTENT_TEST_CASES for parameterized content tests (4 → 1)
- Add ERROR_TEST_CASES for parameterized error tests (2 → 1)
- Add REMOTE_CONTENT_CASES for parameterized remote tests (3 → 1)
- Use shared mock_storage_execute_command fixture
- Use verify_result_structure() helper for common assertions
- Split remote execution into focused tests (content, args, failure)
- Reduce 15 test methods to 11 (~27% reduction)
- Reduce ~50 lines while maintaining full test coverage

☝️ Depends on #101 to merge first

## Summary by CodeRabbit

* **Tests**
  * Expanded test coverage for storage tools, including parameterized test cases for sorting, filtering, and remote execution scenarios.
  * Added tests for edge cases such as empty directories, invalid parameters, and permission errors.
  * Enhanced test utilities to validate result structures across various content types and path resolution scenarios.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: tests/tools/test_storage.py [hunk 1]
```diff
@@ -288,208 +289,201 @@ async def test_list_block_devices_exception_handling(self, mock_storage_execute_
 
 @pytest.mark.skipif(sys.platform != "linux", reason="requires GNU version of coreutils/findutils")
 class TestListDirectories:
+    """Test suite for list_directories tool."""
+
+    # Test data for sorting tests - (order_by, sort, dir_specs, expected_entries)
+    SORT_TEST_CASES = [
```
Comment (r0x0d) [inline tests/tools/test_storage.py:295]: I guess this is duplicated.. Can we move this to be outside the class and only add the cases that are related to the test itself?

Comment (major) [reply tests/tools/test_storage.py:295]: That makes sense. This one is stuck behind two other PRs which is also waiting on #97 to get in, so I'll revisit this once the chain starts moving...

Comment (r0x0d) [reply tests/tools/test_storage.py:295]: that was the only thing I spotted in this PR that I would like to have changed. Other than that, looks fine for me
